roxctl/netpol/generate/generate.go

```diff
@@ -105,9 +119,17 @@ func (cmd *netpolGenerateCmd) validate() error {
 		}
 	}
 
-	portErrs := validation.IsValidPortNum(int(cmd.Options.DNSPort))
-	if len(portErrs) > 0 {
-		return fmt.Errorf("illegal port number: %s", portErrs[0])
+	if cmd.dnsPortName != "" {
+		portErrs := validation.IsValidPortName(cmd.dnsPortName)
+		if len(portErrs) > 0 {
+			return errox.InvalidArgs.Newf("illegal port name: %s", portErrs[0])
+		}
+	} else if cmd.Options.DNSPort != "" { // user set a value which could be interpreted as an integer
```

Comment:
I think, cobra has other means to check whether a flag was set by the user. This check may not tell whether the flag was not used from the user setting that arbitrarily to empty string.

Note that behavior it those two cases should be different. If user provides empty string, we should return input validation error and ask the user to do better. In case where this flag is not used we would then apply the default value for the port.

Reply:
Fixed as suggested.